components/src/Modal/Modal.js

```diff
@@ -0,0 +1,308 @@
+
+    return <PrimaryButtonComponent {...primaryButton}>{primaryButton.label}</PrimaryButtonComponent>;
+  }
+
+  getSecondaryButtons() {
+    const { secondaryButtons } = this.props;
+
+    if (!Array.isArray(secondaryButtons)) {
+      return null;
+    }
+
+    return secondaryButtons.map(button => (
+      <Button {...button} key={button.label}>
+        {button.label}
+      </Button>
+    ));
+  }
+
+  getModalButtons() {
+    const { buttonAlignment } = this.props;
+
+    if (buttonAlignment === "spaced") {
+      return (
+        <React.Fragment>
+          {this.getSecondaryButtons()}
+          {this.getPrimaryButton()}
+        </React.Fragment>
+      );
+    } else {
+      return (
+        <React.Fragment>
+          {this.getPrimaryButton()}
+          {this.getSecondaryButtons()}
+        </React.Fragment>
+      );
+    }
+  }
+
+  modalHasHeader() {
+    const { onRequestClose, title } = this.props;
+    return onRequestClose || title;
+  }
+
+  modalHasFooter() {
+    const { primaryButton, secondaryButtons } = this.props;
+    return primaryButton || secondaryButtons;
+  }
+
+  render() {
+    const {
+      isOpen,
+      children,
+      title,
+      primaryButton,
+      secondaryButtons,
+      type,
+      onRequestClose,
+      onAfterOpen,
+      buttonAlignment,
+      shouldFocusAfterRender,
+      shouldReturnFocusAfterClose,
+      ariaLabel,
+      ariaDescribedBy,
+      portalClassName,
+      overlayClassName,
+      className,
+      maxWidth,
+      appElement,
+      ariaHideApp
+    } = this.props;
+    return (
+      <StyledReactModal
+        maxWidth={maxWidth}
+        contentLabel={ariaLabel}
+        onRequestClose={onRequestClose}
+        onAfterOpen={onAfterOpen}
+        shouldFocusAfterRender={shouldFocusAfterRender}
+        shouldReturnFocusAfterClose={shouldReturnFocusAfterClose}
+        isOpen={isOpen}
+        portalClassName={portalClassName}
+        overlayClassName={overlayClassName}
+        className={className}
+        aria={{
+          labelledby: title ? "modal-title" : undefined,
+          describedby: ariaDescribedBy
+        }}
+        shouldCloseOnOverlayClick
+        shouldCloseOnEsc
+        style={{
+          overlay: overlayStyle
+        }}
+        appElement={appElement}
+        ariaHideApp={ariaHideApp}
+      >
+        <PreventBodyElementScrolling>
+          {this.modalHasHeader(onRequestClose, title) && (
+            <ModalHeader hasCloseButton={onRequestClose}>
+              {title ? (
+                <SectionTitle id="modal-title" mb="none">
+                  {title}
+                </SectionTitle>
+              ) : (
+                <div style={{ height: theme.space.x4 }} />
+              )}
+              {onRequestClose && <ModalCloseButton onClick={onRequestClose} />}
+            </ModalHeader>
+          )}
+          <ModalContent>
+            <InnerModalContent>{children}</InnerModalContent>
```

Comment:
`SomeComponent` and `InnerSomeComponent` is a design smell.

Why do we need `ModalContent` and `InnerModalContent`?
Is there a way to handle `InnerModalContent` inside `ModalContent`?